dotnet/src/Connectors/Connectors.Memory.Weaviate/WeaviateMemoryStore.cs

```diff
@@ -48,6 +49,11 @@ public class WeaviateMemoryStore : IMemoryStore
         DefaultIgnoreCondition = JsonIgnoreCondition.WhenWritingNull
     };
 
+    static WeaviateMemoryStore()
+    {
+        s_jsonSerializerOptions.Converters.Add(new ReadOnlyMemoryConverter());
```

Comment:
@jevgenigeurtsen Great catch, thank you for contribution! Can we add `ReadOnlyMemoryConverter` at place where we initialize `s_jsonSerializerOptions` instead of static constructor? In this case we will follow the pattern used in other places across codebase. 

Also, if you have a chance, we can consider adding a unit tests which will cover serialization/deserialization scenarios, so we won't receive similar problems in the future. Let me know if it's possible for you, otherwise I will join you and add test scenarios as part of this PR. Thanks again!

Reply:
I've updated to code to be aligned with the codebase.

Regarding the tests, there are some integration tests that implicitly cover the serialization/deserialization scenarios. They seem to be disabled at the moment, probably because they require the docker container to be up and running. Is this something we could provide?